### DISCUSSION TITLE
新版本有推荐的效果器搭配吗？

### DISCUSSION THREAD

**Discussion by mrl998** [Blinue/Magpie#880] [General]
看这次更新多了几个系列的新效果器，我还在用原来的A4K上采样+线条修复

**Comment by letterk** [comment]
我感觉动漫2D游戏, A4K线条重建和上采样是最强的, 然后用fsr填满屏幕. 别的要么效果差, 要么感知弱, 就a4k感知是最强的

**Comment by BLACKCHARGE** [comment]
A4k的restore按需使用，而且要放在上采样前面，看素材有没有压缩，压过比较糊就用，否则如果是原生的就别用；上采样一般就是cunny现在更好些

**Comment by AyaRu97** [comment]
現在我是這樣的配置，感覺已經是最還原了
顯示卡 : RX6900XT
1080p > 4k : 直接 CuNNy-8x8C-NVL + AdaptiveSharpen 0.2 
其實是可以用CuNNy-4(8)x16C-NVL..但不知為何用了有時會莫名其妙地輕微地卡頓一下，所以還是用8x8就好了...
720P > 4k : 
```
            "effects": [
//先放大兩倍(1280x720 > 2560x1440)
                {
                    "name": "Anime4K\\Anime4K_Upscale_VL"
                },
//稍微銳化一下
                {
                    "name": "Sharpen\\AdaptiveSharpen",
                    "parameters": {
                        "curveHeight": 0.19999998807907104
                    }
                },
//縮小(2560x1440 > 1920x1080)(用來減輕顯卡負荷)
                {
                    "name": "Bicubic",
                    "scalingType": 0,
                    "scale": {
                        "x": 0.75,
                        "y": 0.75
                    },
                    "parameters": {
                        "paramB": 0.0,
                        "paramC": 0.75
                    }
                },
//再放大兩倍
                {
                    "name": "CuNNy\\CuNNy-8x8C-NVL"
                }
            ]
```
↓↓↓ A4K_Upscale_VL > AdaptiveSharpen > Bicubic > CuNNy-8x8C-NVL ↓↓↓
![bandicam 2024-08-22 02-17-27-143](https://github.com/user-attachments/assets/1d3e81d8-0d0c-4b3a-9e94-77333de81c39)
↓↓↓ 原生放大(720p>4k) ↓↓↓
![bandicam 2024-08-22 02-21-04-268](https://github.com/user-attachments/assets/2fc00f89-f860-4584-a35a-cd5ba9d184c2)
↓↓↓ A4K_Upscale_VL > AdaptiveSharpen > Bicubic > A4K_Upscale_VL ↓↓↓
![bandicam 2024-08-22 02-17-13-124](https://github.com/user-attachments/assets/3ac26d17-1c91-4bd4-ae5d-d86a49b61ed7)
可以看看眼睛的部份，用了兩次a4k的話，顏色好像變得更加光亮了...個人不太喜歡這種改變...

**Reply by letterk** [reply]
末端是放大了也看不出来A4K_Upscale_VL和CuNNy-8x8C-NVL有啥差别

**Reply by mrl998** [reply]
我的感觉也是A4K出来的饱和度会比CuNNy的略高一点，不过说不准哪个更接近原图

**Comment by 2ji3150** [comment]
CuNNy是不是有新的模型？https://github.com/funnyplanter/CuNNy/tree/master/magpie/normal

**Comment by CCXXXI** [comment]
1. 先确定上采样。比如 720p 到 4k 可以用 A4K 的 GAN_x3，也可以用两次 x2 的采样。我测下来是用两次 [CuNNy](https://github.com/funnyplanter/CuNNy) 的新模型最舒适。
2. 然后加不改变分辨率的处理。个人偏好 A4K 的 restore。位置上越往前越快、效果越明显（明显≠好），优先考虑加在上采样前（具体看原图是否有 [瑕疵](https://vcb-s.com/archives/4738)，动画几乎都有，部分游戏也有），其次是加在第一次上采样后，如果觉得还不够锐就再加在第二次上采样后。
3. 最后调整下采样的位置，默认是在最后，往前移可以提升性能但可能会降低画质。我觉得放在第二次上采样前最好，提速的同时对画质几乎没影响。

<details><summary>配置文件，供参考</summary>

```json
{
    "scalingModes": [
        {
            "name": "无损 720p",
            "effects": [
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                },
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                },
                {
                    "name": "CuNNy new\\Downscale",
                    "scalingType": 1,
                    "scale": {
                        "x": 0.5,
                        "y": 0.5
                    }
                },
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                }
            ]
        },
        {
            "name": "无损 1080p",
            "effects": [
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                },
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                }
            ]
        },
        {
            "name": "有损 720p",
            "effects": [
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                },
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                },
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                },
                {
                    "name": "CuNNy new\\Downscale",
                    "scalingType": 1,
                    "scale": {
                        "x": 0.5,
                        "y": 0.5
                    }
                },
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                }
            ]
        },
        {
            "name": "有损 1080p",
            "effects": [
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                },
                {
                    "name": "CuNNy new\\CuNNy-8x32-NVL"
                },
                {
                    "name": "Anime4K\\Anime4K_Restore_UL"
                }
            ]
        }
    ]
}
``` 

</details>

**Reply by BLACKCHARGE** [reply]
游戏需不需要restore要具体看情况，多数游戏素材是原生的很锐利确实不需要，但是部分游戏尤其是有些live2D的游戏包括立绘在内的素材本身经过了插值之类的处理，模糊和锯齿很严重，这种情况光放大不解决问题，还是要靠restore

**Reply by CCXXXI** [reply]
> 游戏需不需要restore要具体看情况，多数游戏素材是原生的很锐利确实不需要，但是部分游戏尤其是有些live2D的游戏包括立绘在内的素材本身经过了插值之类的处理，模糊和锯齿很严重，这种情况光放大不解决问题，还是要靠restore

确实，而且得在放大前处理，不然把瑕疵一并放大之后就难修了

**Reply by darkblue112233** [reply]
用了你这配置字体比只用A4K字体清晰太多了，我只用Anime4k字体太**模糊**，我想问下用这些模式前需要把游戏的高DPI缩放替代改成应用程序吗，还是说默认不要改比较好

**Reply by CCXXXI** [reply]
> 用了你这配置字体比只用A4K字体清晰太多了，我只用Anime4k字体太**模糊**

主要是 CuNNy 的效果，Anime4K 的 Upscale 会把文字边缘弄变形。

> 我想问下用这些模式前需要把游戏的高DPI缩放替代改成应用程序吗，还是说默认不要改比较好

如果用默认的 Graphics Capture 模式的话，要改。这个不是特定配置的要求，是 [Magpie 需要这样才能捕获原图](https://github.com/Blinue/Magpie/wiki/%E6%8D%95%E8%8E%B7%E6%96%B9%E5%BC%8F%E5%AF%B9%E6%AF%94)。

**Comment by windom86** [comment]
有大佬推荐一下3d游戏怎么配置吗？找到的都是2d
